enhancements/agent-installer/automated-workflow.md

```diff
@@ -0,0 +1,491 @@
+15. On node0, the startup script in the image recognizes that it is on
+    node0 and launches the assisted service and the create-cluster
+    service to drive the deployment.
+16. On node0, the create-cluster service waits for the assisted
+    service to start, then creates an InfraEnv using the API.
+17. On node0, the create-cluster service waits for all of the known
+    agents to be registered with the service.
+18. On all nodes (including node0), the startup script in the image
+    connects to the assisted service on node0 to find the InfraEnv ID
+    for the cluster being deployed (there will only be 1). After it
+    has the ID, the startup script launches the assisted agent with
+    that ID and configured to connect to the assisted service on
+    node0.
+19. All of the agents register their hosts with the assisted service.
+20. The assisted service performs its standard validations for all of the
+    hosts.
+21. On node0, the create-cluster service sees that all known agents
+    have registered and the hosts have passed validation. It uses the
+    assisted service API to trigger the deployment to proceed.
+22. The assisted service runs the OpenShift installer and performs the
+    cluster deployment in the usual way, except that the nodes will
+    use the ISOs they booted from to install RHCOS instead of fetching
+    a new one from the assisted service.
+23. The `openshift-install wait-for install-complete` process sees the
+    OpenShift API become available and starts using it to watch the
+    cluster complete deployment, combining the information provided by
+    `ClusterVersion` and `ClusterOperator` resources with the assisted
+    service progress API.
+24. On node0, when all of the other hosts have been deployed and
+    bootstrapping is complete, the agent reboots node0.
+25. The `openshift-install wait-for install-complete` process loses
+    the connection to the assisted service REST API and starts relying
+    entirely on the OpenShift API for data.
+26. The `openshift-install wait-for install-complete` process sees
+    that the `ClusterVersion` API in the cluster shows that the
+    deployment has completed, and reports success then exits.
+27. The orchestration tool may take steps to clean up (removing the
+    ISO, disconnecting it from the boot sequence for the VMs or BMCs,
+    etc.).
+
+TODO: Write the steps describing what happens when create-cluster sees
+      that a host has failed validation.
+
+### User Stories
+
+- As an OpenShift infrastructure owner, I need to be able to integrate
+  the installation of my first on-premises OpenShift cluster with my
+  automation flows and tools.
+- As an OpenShift infrastructure owner, I must be able to provide the
+  definition of the cluster I want to deploy as input to the tool.
+- As an OpenShift Infrastructure owner, I must be able to retrieve the
+  validation errors from an in-progress or failed deployment in a
+  programmatic way.
```

Comment:
I think defining what "in a programmatic way" means may be helpful (perhaps in the TODO about the failed-validation path above), e.g as an Infrastructure owner with some existing orchestration tool (ansible, etc) how will I automate this flow, and detect and respond to error conditions?

Reply:
calling the openshift-install subcommands would be the easier, since they will know how to deal with both REST APIs. Alternatively, doing the monitoring directly using the REST APIs

Reply:
See the variations listed above starting around line 331.